### PR TITLE
Community charms in search results

### PR DESCRIPTION
**Pull request by Lukewh**
Fixes https://github.com/CanonicalLtd/jujucharms.com/issues/454 in the GUI. SF to follow.

### CODE REVIEW

**Comment by Lukewh** [comment]
## QA
- Search for 'term' you should get 3 results under the 'Community' header.
- Search for 'kibana' you should get 1 result under 'Recommended' and a 'Show community results' button.
- Clicking the 'Show community results' button should cause the button to change from 'show' -> 'hide' and will show an additional 14 charms under the 'Community' header.

File: jujugui/static/gui/src/app/components/search-results/search-results.js [hunk 1]
```diff
@@ -295,10 +295,12 @@ YUI.add('search-results', function(Y) {
     */
     generateState: function(nextProps) {
       var state = {
-        activeComponent: nextProps.activeComponent || 'loading'
+        activeComponent: nextProps.activeComponent || 'loading',
+        showCommunity: nextProps.showCommunity || false
       };
       var currentType = nextProps.type !== undefined ?
           nextProps.type : this.props.type;
+      state.currentType = currentType;
```
Comment (Lukewh) [inline jujugui/static/gui/src/app/components/search-results/search-results.js]: I need to remove this. It's not used.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7248/

File: jujugui/static/gui/src/app/components/search-results/search-results.js [hunk 2]
```diff
@@ -467,6 +469,59 @@ YUI.add('search-results', function(Y) {
...
+      </div>);
+    },
+
+    _toggleCommunityResults: function() {
+      let state = this.state;
+      state.showCommunity = !this.state.showCommunity;
+      this.setState(this.generateState(state));
+    },
+
+    _generateCommunityResults: function(community, hasPromulgated) {
```
Comment (Lukewh) [inline jujugui/static/gui/src/app/components/search-results/search-results.js]: Docstring

File: jujugui/static/gui/src/app/components/search-results/search-results.js [hunk 3]
```diff
@@ -467,6 +469,59 @@ YUI.add('search-results', function(Y) {
...
+          {promulgated.map(item =>
+            <juju.components.SearchResultsItem
+              changeState={this.props.changeState}
+              key={item.storeId}
+              item={item} />)}
+        </ul>
+      </div>);
+    },
+
+    _toggleCommunityResults: function() {
```
Comment (Lukewh) [inline jujugui/static/gui/src/app/components/search-results/search-results.js]: Docstring

File: jujugui/static/gui/src/app/components/search-results/search-results.js [hunk 4]
```diff
@@ -467,6 +469,59 @@ YUI.add('search-results', function(Y) {
       window.history.back();
     },
 
+    _generatePromulgatedResults: function(promulgated) {
```
Comment (Lukewh) [inline jujugui/static/gui/src/app/components/search-results/search-results.js]: Docstring

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7250/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7251/

**Comment by frankban** [comment]
This also is a fix for https://github.com/juju/juju-gui/issues/2435 correct?

**Comment by Lukewh** [comment]
Correct @frankban 

:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**